# https://bugs.gentoo.org/show_bug.cgi?id=250306
# Allow to build xulrunner with libxul in place of libxul-embedding
--- a/configure.in	2008-09-22 16:43:15.000000000 +0300
+++ b/configure.in	2009-03-16 20:32:14.000000000 +0200
@@ -88,13 +88,16 @@
 *) AC_MSG_ERROR([$gecko_cv_gecko is not supported])
 esac
 
-if test $GECKO = libxul-embedding; then
-	GECKO_CFLAGS=`pkg-config --cflags libxul-embedding '>=' $gecko_min_version $gecko_cv_extra_pkg_dependencies`
-	GECKO_LIBS=`pkg-config --libs libxul-embedding '>=' $gecko_min_version $gecko_cv_extra_pkg_dependencies`
-else
-	GECKO_CFLAGS=`pkg-config --cflags ${GECKO}-xpcom '>=' $gecko_min_version $GECKO_EXTRA_PKG_DEPENDENCIES`
-	GECKO_LIBS=`pkg-config --libs ${GECKO}-xpcom '>=' $gecko_min_version $GECKO_EXTRA_PKG_DEPENDENCIES`
-fi
+case "$GECKO" in
+libxul*)
+	GECKO_CFLAGS=`pkg-config --cflags ${GECKO} '>=' $gecko_min_version $gecko_cv_extra_pkg_dependencies`;
+	GECKO_LIBS=`pkg-config --libs ${GECKO} '>=' $gecko_min_version $gecko_cv_extra_pkg_dependencies`;
+	;;
+*)
+	GECKO_CFLAGS=`pkg-config --cflags ${GECKO}-xpcom '>=' $gecko_min_version $GECKO_EXTRA_PKG_DEPENDENCIES`;
+	GECKO_LIBS=`pkg-config --libs ${GECKO}-xpcom '>=' $gecko_min_version $GECKO_EXTRA_PKG_DEPENDENCIES`;
+	;;
+esac
 
 AC_SUBST([GECKO_CFLAGS])
 AC_SUBST([GECKO_LIBS])
